### PR TITLE
support overriding task run, and update trigger notification

### PR DESCRIPTION
**Pull request by shahzzzam**
Please ignore all code under src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/sdk
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/task.py [hunk 1]
```diff
@@ -612,15 +631,32 @@ def acr_task_run(cmd,
...
+    import base64
+    if update_trigger_token:
+        update_trigger_token = base64.b64encode(update_trigger_token.encode()).decode()
+
+    task_id=get_task_id_from_task_name(cmd.cli_ctx, resource_group_name, registry_name, task_name)
+
+    override_task_step_properties = OverrideTaskStepProperties(
+        context_path=context,
+        file=file,
+        arguments=(arguments if arguments else []) + (secret_arguments if secret_arguments else []),
```
Comment (northtyphoon) [inline src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/task.py:647]: >arguments [](start = 8, length = 9)

Do we merge argument and secret_arguments in the override case?

File: src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_utils.py [hunk 2]
```diff
@@ -12,6 +12,7 @@
     REGISTRY_RESOURCE_TYPE,
     ACR_RESOURCE_PROVIDER,
     STORAGE_RESOURCE_TYPE,
+    TASK_ID_TEMPLATE,
```
Comment (northtyphoon) [inline src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_utils.py]: >TASK_ID_TEMPLATE [](start = 4, length = 16)

Can you rename ID to RESOURCE_ID?

File: src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_constants.py [hunk 3]
```diff
@@ -39,7 +40,8 @@ def get_valid_os(cmd):
 
 
 def get_valid_architecture(cmd):
-    Architecture = cmd.get_models('Architecture')
+    from .sdk.models import Architecture
+    # Architecture = cmd.get_models('Architecture')
```
Comment (northtyphoon) [inline src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_constants.py:44]: Architecture is commented

File: src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/task.py (hunk 1 above)
Comment (shahzzzam) [reply src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/task.py:647]: Yes, we always merge `arguments/secret_arguments` and `setValue/setSecret`. Same for override

File: src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_constants.py (hunk 3 above)
Comment (shahzzzam) [reply src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_constants.py:44]: Yes, waiting for the new SDK to be merged. Once thats done, I will not use local SDK.

**Pull request closed without merging**